src/lib/autodiscovery/autodiscovery.py

```diff
@@ -18,6 +20,66 @@
 discovered_resource_type = os.environ.get("AUTODISCOVERY_RESOURCE_TYPE", "gce_instance")
 
 
+async def send_metadata(missing_metrics_list: List[Metric], gcp_session: ClientSession, token: str):
+    to_send = []
+
+    for metric in missing_metrics_list:
+        metric_name = metric.dynatrace_name[0:250]
+        display_name = (
+            "[Autodiscovered] " + metric.name[0:280] if len(metric.name) >= 1 else metric_name
+        )
+        description = (
+            metric.description[0:65535]
+            if len(metric.description) >= 1
+            else "Unspecified Metric Description"
+        )
+        unit = metric.unit
+        to_send.append(
+            {
+                "scope": f"metric-{metric_name}",
+                "schemaId": "builtin:metric.metadata",
+                "value": {
+                    "displayName": f"{display_name}",
+                    "description": f"{description}",
+                    "unit": f"{unit}",
+                    "dimensions": [],
+                    "tags": [],
+                    "sourceEntityType": "cloud:gcp:gce_instance",
+                },
+            }
+        )
+
+    dynatrace_api_key = (await fetch_dynatrace_api_key(gcp_session, config.project_id(), token),)
+    dynatrace_url = (await fetch_dynatrace_url(gcp_session, config.project_id(), token),)
+    dt_url = f"{dynatrace_url[0].rstrip('/')}/api/v2/settings/objects"
+
+    async with init_dt_client_session() as dt_session:
+        response = await dt_session.post(
+            url=dt_url,
+            headers={
+                "Authorization": f"Api-Token {dynatrace_api_key[0]}",
```

Comment:
check how it is done via ENV variables - you will get a normal element, not an array. + get rid of () to avoid it as mentioned before